### PR TITLE
Feature | Breadcrumbs

### PR DESCRIPTION
**Pull request by ChazUK**
We're going to have to fiddle around with the routes and titles for pages to make sure we aren't receiving any double pages, and the nesting is working correctly

### CODE REVIEW

**Comment by ChazUK** [comment]
[(3) Breadcrumb Component](https://trello.com/c/VtvbKp3X/160-3-breadcrumb-component)

File: src/app/shared/components/breadcrumbs/breadcrumbs.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<ng-container *ngIf="display">
+  <div class="govuk-breadcrumbs">
+    Go back to:
+    <ol class="govuk-breadcrumbs__list">
```
Comment (omarCreativeDev) [inline src/app/shared/components/breadcrumbs/breadcrumbs.component.html]: how comes this is an `<ol>`? wouldn't this render numbers on the list items or are those being hidden for assessibility?

File: src/app/core/services/nested-routes.service.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        map(() => this.activatedRoute)
+      )
+      .subscribe(route => {
+        const routes = this.getNestedRoutes(route);
+        this._routes$.next(routes);
+      });
+  }
+
+  private getNestedRoutes(route: ActivatedRoute, url: string = '', routes: IRoute[] = []): IRoute[] {
+    const ROUTE_DATA_BREADCRUMB = 'title';
```
Comment (omarCreativeDev) [inline src/app/core/services/nested-routes.service.ts]: why is this variable not camelCased?

File: src/app/core/components/header/header.component.html [hunk 3]
```diff
@@ -28,7 +28,7 @@
         <ng-container *ngIf="isLoggedIn()">
           <li class="govuk-header__navigation-item">{{ fullname }}</li>
           <li class="govuk-header__navigation-item">
-            <a routerLink="/account-management/your-account" class="govuk-header__link">Account</a>
+            <a routerLink="/account-management" class="govuk-header__link">Account</a>
```
Comment (omarCreativeDev) [inline src/app/core/components/header/header.component.html]: this change makes sense, keep things consistent with the other modules

File: src/app/shared/components/breadcrumbs/breadcrumbs.component.html (hunk 1 above)
Comment (ChazUK) [reply src/app/shared/components/breadcrumbs/breadcrumbs.component.html]: I took the start of this markup from the GOV.UK Framework https://design-system.service.gov.uk/components/breadcrumbs/

I think the logic why it's an `ol` and not a `ul` is that there is a hierarchy, level 1, level 2, level 3 and so on.

File: src/app/core/services/nested-routes.service.ts (hunk 2 above)
Comment (ChazUK) [reply src/app/core/services/nested-routes.service.ts]: yep, I can update this or even remove it.

File: src/app/core/components/header/header.component.html (hunk 3 above)
Comment (ChazUK) [reply src/app/core/components/header/header.component.html]: There might be a few other places we need to check this too.

File: src/app/core/services/nested-routes.service.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/core/services/nested-routes.service.ts]: if you can rename it to camelCase that would be super

File: src/app/core/components/header/header.component.html (hunk 3 above)
Comment (omarCreativeDev) [reply src/app/core/components/header/header.component.html]: I will keep an eye out for this once i jump on to the user accounts ticket

File: src/app/core/services/nested-routes.service.ts (hunk 2 above)
Comment (ChazUK) [reply src/app/core/services/nested-routes.service.ts]: done

**Pull request merged**